thirdeye/thirdeye-pinot/src/main/resources/assets/javascript/models/AnomalyResultModel.js

```diff
@@ -30,6 +30,22 @@ function AnomalyResultModel() {
 }
 
 AnomalyResultModel.prototype = {
+  hasSameParams(params) {
+    if (!params.anomaliesSearchMode || !this.anomaliesSearchMode) {
+      return false;
+    }
+    const paramKeys = Object.keys(params);
+    for (let index in paramKeys){
```

Comment:
Yeah, this code needs comment as I can see it's a bit hard to understand what the intention is. Don't think .every would work for me here, but I'll refactor and make it cleaner. Thank you!